### PR TITLE
[DEV-1908] target_table: add service, routes and models

### PR DESCRIPTION
**Pull request by jevonyeoh**
## Description
This PR adds the basic service, routes and models for target_table, which will be used to help us fully materialize targets in the warehouse.

## Related Issue
https://featurebyte.atlassian.net/browse/DEV-1908

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/routes/target_table/controller.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        target_service: TargetService,
+    ):
+        super().__init__(service=target_table_service, preview_service=preview_service)
+        self.feature_store_service = feature_store_service
+        self.observation_table_service = observation_table_service
+        self.entity_validation_service = entity_validation_service
+        self.task_controller = task_controller
+        self.target_service = target_service
+
+    async def create_target_table(
```
Comment (jevonyeoh) [inline featurebyte/routes/target_table/controller.py:55]: this function is somewhat similar to its `historical_feature_table` counterpart. i'll refactor in a follow-up - https://featurebyte.atlassian.net/browse/DEV-1910

mostly copied over for now to get this up and running quickly.

File: featurebyte/worker/task/target_table.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    TargetTableTask creates a TargetTable by computing historical features
+    """
+
+    payload_class = TargetTableTaskPayload
+
+    async def execute(self) -> Any:
+        """
+        Execute TargetTableTask
+        """
+        # TODO: fully implement in follow-up
```
Comment (jevonyeoh) [inline featurebyte/worker/task/target_table.py:29]: will implement in follow-up to try to consolidate w/ historical feature tables as well

File: featurebyte/service/target_table.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+from featurebyte.storage import Storage
+
+
+class TargetTableService(BaseMaterializedTableService[TargetTableModel, TargetTableModel]):
+    """
+    TargetTableService class
+    """
+
+    document_class = TargetTableModel
+    materialized_table_name_prefix = MaterializedTableNamePrefix.HISTORICAL_FEATURE_TABLE
```
Comment (YS-L) [inline featurebyte/service/target_table.py:28]: The prefix here should be updated.

File: featurebyte/service/target_table.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+from featurebyte.enum import MaterializedTableNamePrefix
+from featurebyte.models.base import FeatureByteBaseDocumentModel
+from featurebyte.models.target_table import TargetTableModel
+from featurebyte.schema.target_table import TargetTableCreate
+from featurebyte.schema.worker.task.target_table import TargetTableTaskPayload
+from featurebyte.service.materialized_table import BaseMaterializedTableService
+from featurebyte.storage import Storage
+
+
+class TargetTableService(BaseMaterializedTableService[TargetTableModel, TargetTableModel]):
```
Comment (YS-L) [inline featurebyte/service/target_table.py:22]: This service also seems almost identical to HistoricalTableService. Is there there a plan to refactor and consolidate?

File: featurebyte/models/target_table.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+"""
+HistoricalFeatureTableModel
```
Comment (YS-L) [inline featurebyte/models/target_table.py:2]: TargetTableModel

File: featurebyte/schema/target_table.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    """
+    Schema for listing targe tables
+    """
+
+    data: List[TargetTableModel]
+
+
+class TargetTableListRecord(BaseMaterializedTableListRecord):
+    """
+    Schema for listing historical feature tables as a DataFrame
```
Comment (YS-L) [inline featurebyte/schema/target_table.py:55]: `target tables`

File: featurebyte/service/target_table.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+            document=FeatureByteBaseDocumentModel(_id=output_document_id, name=data.name),
+        )
+
+        if observation_set_dataframe is not None:
+            observation_set_storage_path = (
+                f"target_table/observation_set/{output_document_id}.parquet"
+            )
+            await storage.put_dataframe(
+                observation_set_dataframe, Path(observation_set_storage_path)
+            )
```
Comment (YS-L) [inline featurebyte/service/target_table.py:71]: Not specific to this PR, but we should check whether there is a need to cleanup the temp storage on task completion.

**Review by YS-L — APPROVED** [review]

File: featurebyte/service/target_table.py (hunk 7 above)
Comment (jevonyeoh) [reply featurebyte/service/target_table.py:71]: i believe it uses python tempdir under the hood, so hopefully that should clean itself up _eventually_.

**Pull request merged**